components/board/collector.go

```diff
@@ -82,22 +60,21 @@ func newGPIOCollector(resource interface{}, params data.CollectorParams) (data.C
 	}
 
 	cFunc := data.CaptureFunc(func(ctx context.Context, arg map[string]*anypb.Any) (interface{}, error) {
-		var readings []GpioRecord
-		for k := range arg {
-			if gpio, err := board.GPIOPinByName(k); err == nil {
-				value, err := gpio.Get(ctx, data.FromDMExtraMap)
-				if err != nil {
-					// A modular filter component can be created to filter the readings from a component. The error ErrNoCaptureToStore
-					// is used in the datamanager to exclude readings from being captured and stored.
-					if errors.Is(err, data.ErrNoCaptureToStore) {
-						return nil, err
-					}
-					return nil, data.FailedToReadErr(params.ComponentName, gpios.String(), err)
+		var value bool
+		if gpio, err := board.GPIOPinByName(arg["reader_name"].String()); err == nil {
```

Comment:
Maybe pin_name or pin_number? Not actually sure what the input is here

Reply:
When I talked to Alexa, we settled on "reader_name": ____ but I also would be fine with pin_name.

Reply:
SGTM, just wanted to make sure this decision was intentional. Thanks!